src/pages/other-pension-providers/app.ts

```diff
@@ -30,28 +29,42 @@ const get = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult>
     return renderAsHtmlResponse(event, 'template.njk', { session });
 };
 
-const post = (event: APIGatewayProxyEvent): GatewayResult => {
-    return parseForm(event, processForm(event), [form_key]);
-};
+const post = (event: APIGatewayProxyEvent): GatewayResult =>
+    parseForm(event, processForm(event), [otherPensionProvidersKey]);
+
+const isValid = (options: string[], values: string[]) =>
+    values.every((option) => {
+        return options.includes(option);
+    });
 
-const processForm =
+export const processForm =
     (event: APIGatewayProxyEvent) =>
     async (form: Form): Promise<APIGatewayProxyResult> => {
         const errors: ErrorCollection = {};
+
         function renderPageWithErrors() {
             return renderAsHtmlResponse(event, 'template.njk', { form, errors });
         }
 
-        if (!(form[form_key] && includes(valid_options, form[form_key]))) {
-            errors[form_key] = { text: 'Choose either Yes, No or Skip for now' };
+        const otherPensionProvidersArray = form[otherPensionProvidersKey]?.split(',') || [];
+        if (!(form[otherPensionProvidersKey] && isValid(otherPensionProviderOptions, otherPensionProvidersArray))) {
+            errors[otherPensionProvidersKey] = { text: 'Select at least one pension or None' };
             return renderPageWithErrors();
         }
-        await updateSession(event, form);
+
+        const otherPensionProviders = new OtherProviders();
+        otherPensionProviders[otherPensionProvidersKey] = otherPensionProvidersArray;
+
+        await updateSession(event, otherPensionProviders);
         return {
             statusCode: 303,
             headers: {
-                location: '/check-answers',
+                location: '/tasklist',
             },
             body: '',
         };
     };
+
+class OtherProviders {
+    'other-pension-providers': string[];
+}
```

Comment:
As discussed this can be inlined as an object literal of type `Partial<Session>`